### DISCUSSION TITLE
User Search API - Framework filters are not working

### DISCUSSION THREAD

**Discussion by komalm** [Sunbird-Lern/Community#89] [Query] [answered]
Hello @reshmi-nair, @Hari-stackroute
Hope you are doing well!..

**Site Environments Details** - SunbirdEd portal of version 4.4.1
I am referring the documentation of user search API [here](https://lern.sunbird.org/learn/product-and-developer-guide/user-and-org-service/api-documentation/user-management)

We are creating the users using the user-create API.

**- Payload**

```
{
    "params": {},
    "request": {
        "firstName": "Komal",
        "lastName": "CC2 ",
        "channel": "{{channelName}}",
        "userName": "komalcc2",
        "email": "komalcc2@yopmail.com",
        "emailVerified": true,
        "password": "password",
        "roles": [
            "CONTENT_CREATOR",
            "PUBLIC"
        ],
        "framework": {
            "board": [
                "PQR"
            ],
            "medium": [
                "MNO"
            ],
            "gradeLevel": [
                "ABC"
            ],
            "subject": [],
            "subcategory": [
                "Government"
            ],
            "category": [
                "Individual"
            ],
            "city": [
                "Pune"
            ],
            "institution": [],
            "id": [
                "{{frameworkName}}"
            ]
        }
    }
}
```

In the framework, we included a few custom nodes or fields, such as city, category, and sub-category, as mentioned in the above payload.

With the user search API, we now wished to apply a filter.
Payload of serch API Filters include:

```
{
    "request": {
        "query": "",
        "filters": {
            "channel": "{{channelName}}",
            "framework.category":"Individual"
        }
    }
}
```

But that is not working.

Would you kindly assist me with the same?

cc @Renuka-S, @vinod303

**Comment by reshmi-nair** [comment]
@komalm I will check on this and get back

**Reply by BharathwajShankar** [reply, marked as answer]
@komalm you can search framework-related fields through search API by using the below sample request, for example, if you want to search city then it can be,

{
    "request": {
        "query": "",
        "filters": {
            "channel": "{{channelName}}",
            "email": "komalcc2@yopmail.com",
            "framework": {
                "city": [
                    "Pune"
                ]
            }
        }
    }
}
likewise, you can search any framework details with above structure

**Comment by Jayaprakash8887** [comment]
@komalm Pls let us know if the query is resolved. If yes, kindly mark the thread as answered
